addons/account/models/account_move_line.py

```diff
@@ -1921,7 +1930,6 @@ def _create_reconciliation_partials(self):
                 'company': line.company_id,
                 'currency': line.currency_id,
                 'date': line.date,
-                'reconciled': line.reconciled,
```

Comment:
Is this in need to be removed, strictly speaking? It's true that later is not used in _prepare_reconciliation_partials anymore due to the full revert, but... I mean, perhaps a customization may need it. I think would be nice keeping it.

Reply:
@MiquelRForgeFlow If you line has no balance but an amount in foreign currency, the line is considered as an exchange difference line here:
https://github.com/odoo/odoo/blob/375aa97f36356501fadd86e1be2fc44a0b608093/addons/account/models/account_move_line.py#L1692

...in `_prepare_reconciliation_single_partial`.

To me, adding `reconciled` is completely wrong because the reconciliation is already raising an error in that case:
https://github.com/odoo/odoo/blob/375aa97f36356501fadd86e1be2fc44a0b608093/addons/account/models/account_move_line.py#L2297

You can't expect any reconciliation in that case. The meaning of this boolean is "There is nothing left to do with this line".